### DISCUSSION TITLE
First, Last, Next and Previous Entries for a Given List of Entries

### DISCUSSION THREAD

**Discussion by richhayler** [craftcms/cms#11343] [Q&A] [answered]
On an entry page, I would like to return the **entry title** and **url** for the first, last, previous and next entries (in the context of the current entry) for a given list of entries.

I can use the following twig code to get the previous and next information but there doesn't seem to be a similar method to do the same for returning the first and last entries.

Am I missing something? It would be nice to have the option to use something like, `getFirst()` and `getLast()` as a shorthand approach in contrast to what is described in the docs: [https://craftcms.com/docs/4.x/dev/tags.html#the-pageinfo-variable](https://craftcms.com/docs/4.x/dev/tags.html#the-pageinfo-variable)

```
{% set criteria = {section: 'mySection', orderBy: 'postDate desc', limit: 10 } %}

{% set prev  = entry.getPrev(criteria) %}
{% set next  = entry.getNext(criteria) %}

 {{ prev ? prev.title ~ '-' ~ prev.url }}
 {{ next ? next.title ~ '-' ~ next.url }}

{# It would be nice to do this.... #}

{% set first  = entry.getFirst(criteria) %}
{% set last  = entry.getLast(criteria) %}

 {{ first ? first.title ~ '-' ~ first.url }}
 {{ last ? last.title ~ '-' ~ last.url }}


```

**Comment by brandonkelly** [comment, marked as answer]
If you create an entry query rather than a parameter array, you can call its `one()` method to get the first matching entry, and then clone it and call `inReverse()` + `one()` to get the last entry:

```twig
{% set query = craft.entries()
  .section('mySection')
  .orderBy('postDate desc')
  .limit(10) %}

{% set prev = entry.getPrev(query) %}
{% set next = entry.getNext(query) %}
{% set first = query.one() %}
{% set last = clone(query).inReverse().one() %}
```

**Reply by richhayler** [reply]
Thank you @brandonkelly! I really appreciate you answering my question. I completely missed `inReverse()` so thank you for pointing this out :) and providing a super simple answer to my problem.

**Reply by richhayler** [reply]
I also wanted to share a practical example for using the above.... creating a simple pagination loop control (on an entry page), that shows the name and the link for the previous and next entries. When navigating forward (next) to the last entry, the next title and link are replaced with the title and link to the first entry. When navigating to the begining (last previous entry) then the title and link to the last entry is display.

```
<a href="{{ prev ? prev.url : last.url }}">{{ prev ? prev.title : last.title }}</a>
<a href="{{ next ? next.url : first.url }}">{{ next ? next.title : first.title }}</a>
```
